xray/test/test_variable.py

```diff
@@ -197,6 +197,12 @@ def test_pandas_data(self):
         v = self.cls(['x'], pd.Index([0, 1, 2]))
         self.assertEqual(v[0].values, v.values[0])
 
+    def test_pandas_period_index(self):
+        v = self.cls(['x'], pd.period_range(start='2000', periods=20, freq='B'))
+        self.assertEqual(v[0], pd.Period('2000', freq='B'))
+        # test repr works #GH645
+        repr(v)
```

Comment:
This actually hits a different codepath that never involves the logic in `core.formatting`. You'll need to put the variable in a Dataset if you want to see that.